rootston/output.c

```diff
@@ -512,6 +532,31 @@ static void render_output(struct roots_output *output) {
 
 damage_finish:
 	pixman_region32_fini(&damage);
+
+	// Send frame done events to all surfaces
+	if (output->fullscreen_view != NULL) {
+		if (wlr_output->fullscreen_surface == output->fullscreen_view->wlr_surface) {
+			// The surface is managed by the wlr_output
+			return;
+		}
+
+		view_for_each_surface(view, surface_send_frame_done, &data);
```

Comment:
I should have posted two separate comments, I think you missed my second point: `view` here is some random view (likely last of `desktop->views`), should it be assigned `output->fullscren_view` before using it here and below?

Reply:
Ah, you're right, good catch.